apps/ewallet_db/test/ewallet_db/transaction_consumption_test.exs

```diff
@@ -0,0 +1,175 @@
+
+      # They are still valid since we haven't made them expired yet
+      assert TransactionConsumption.expired?(t1) == false
+      assert TransactionConsumption.expired?(t2) == false
+      assert TransactionConsumption.expired?(t3) == false
+      assert TransactionConsumption.expired?(t4) == false
+
+      TransactionConsumption.expire_all()
+
+      # Reload all the records
+      t1 = TransactionConsumption.get(t1.id)
+      t2 = TransactionConsumption.get(t2.id)
+      t3 = TransactionConsumption.get(t3.id)
+      t4 = TransactionConsumption.get(t4.id)
+
+      # Now t1 and t2 are expired
+      assert TransactionConsumption.expired?(t1) == true
+      assert TransactionConsumption.expired?(t2) == true
+      assert TransactionConsumption.expired?(t3) == false
+      assert TransactionConsumption.expired?(t4) == false
+    end
+
+    test "sets the expired_at field" do
+      now = NaiveDateTime.utc_now()
+      t = insert(:transaction_consumption, expiration_date: NaiveDateTime.add(now, -60, :seconds))
+      TransactionConsumption.expire_all()
+      t = TransactionConsumption.get(t.id)
+
+      assert TransactionConsumption.expired?(t) == true
+      assert t.expired_at != nil
+    end
+  end
+
+  describe "all_active_for_request" do
+    test "it returns all pending and confirmed consumptions for the given request" do
+      request = insert(:transaction_request)
+      consumption_1 = insert(:transaction_consumption,
+        transaction_request_id: request.id,
+        status: "pending"
+      )
+      consumption_2 = insert(:transaction_consumption,
+        transaction_request_id: request.id,
+        status: "confirmed"
+      )
+      _consumption_3 = insert(:transaction_consumption,
+        transaction_request_id: request.id,
+        status: "failed"
+      )
+      _consumption_4 = insert(:transaction_consumption,
+        transaction_request_id: request.id,
+        status: "expired"
+      )
+      consumption_5 = insert(:transaction_consumption,
+        transaction_request_id: request.id,
+        status: "confirmed"
+      )
+
+      consumptions = TransactionConsumption.all_active_for_request(request.id)
+
+      assert length(consumptions) == 3
+      assert consumption_1 in consumptions == true
+      assert consumption_2 in consumptions == true
+      assert consumption_5 in consumptions == true
+    end
+  end
+
+  describe "insert/1" do
+    test_insert_generate_uuid TransactionConsumption, :id
+    test_insert_generate_timestamps TransactionConsumption
+    test_insert_prevent_blank TransactionConsumption, :amount
+    test_insert_prevent_blank TransactionConsumption, :idempotency_token
+    test_insert_prevent_blank TransactionConsumption, :transaction_request_id
+    test_insert_prevent_blank TransactionConsumption, :balance_address
+    test_insert_prevent_blank TransactionConsumption, :minted_token_id
+
+    test "sets the status to 'pending'" do
+      inserted = insert(:transaction_consumption)
```

Comment:
**Request:** Maybe more accurate to insert via `TransactionConsumption.insert/1`? Otherwise this tests the factory's data